src/sapling/sapling_validation.cpp

```diff
@@ -16,6 +16,144 @@
 
 namespace SaplingValidation {
 
+// Verifies that Shielded txs are properly formed and performs content-independent checks
+bool CheckTransaction(const CTransaction& tx, CValidationState& state, CAmount& nValueOut, bool fIsSaplingActive)
+{
+
+    // v1 must not have shielded data.
+    if (tx.nVersion < CTransaction::SAPLING_VERSION && tx.hasSaplingData()) {
+        return state.DoS(100, error("%s: Not Sapling version with Sapling data", __func__ ),
+                         REJECT_INVALID, "bad-txns-form-not-sapling");
+    }
+
+    // From here, all of the checks are done in +v2 transactions.
+
+    // if the tx has shielded data, cannot be a coinstake, coinbase, zcspend, p2csOut and zcmint
+    bool hasSaplingData = tx.hasSaplingData();
+    if (hasSaplingData) {
+        if (tx.IsCoinStake() || tx.IsCoinBase() || tx.HasZerocoinSpendInputs() || tx.HasP2CSOutputs() ||
+            tx.HasZerocoinMintOutputs())
+            return state.DoS(100, error("%s: Sapling version with invalid data", __func__),
+                             REJECT_INVALID, "bad-txns-invalid-sapling");
+    }
+
+    // +v2, if the tx has no shielded data, return true. No check needed.
+    if (!hasSaplingData) {
+        return true;
+    }
+
+    // If the v5 upgrade was not enforced, then let's not perform any check
+    if (!fIsSaplingActive) {
+        return state.DoS(100, error("%s: Sapling not activated", __func__),
+                         REJECT_INVALID, "bad-txns-invalid-sapling-act");
+    }
+
+    // Upgrade enforced, basic version rules passing, let's check it
+    return CheckTransactionWithoutProofVerification(tx, state, nValueOut);
+}
+
+bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidationState& state, CAmount& nValueOut)
+{
+    // Basic checks that don't depend on any context
+    const Consensus::Params& consensus = Params().GetConsensus();
+    /**
+     * 1. The consensus rule checks to see if tx.Version falls within the range:
+     *        tx.nVersion < CTransaction::SAPLING_VERSION
+     */
+    if (tx.nVersion < CTransaction::SAPLING_VERSION) {
+        return state.DoS(100, error("%s: version too low", __func__ ),
+                         REJECT_INVALID, "bad-txns-version-not-sapling");
+    }
```

Comment:
yes, good catch. I think that did it because at the beginning there was no `CheckTransaction`. It was a long time ago.
Going to add an assertion just to be totally sure that we don't call this method with a another input in the future.